test_pull_requests

```diff
@@ -2089,7 +2089,8 @@ popd
       # First, get the next jenkins build number
       next_build_num = get_next_build(branch, settings)
 
-      pull_id_params = "&BRANCH=#{branch}"
+      branch_param = settings["branch_param"] || "BRANCH"
```

Comment:
With this setup, you would have to repeat the param for test, merge, etc.  You at least should have a fallback to a global config option if these aren't found.